htdocs/core/lib/files.lib.php

```diff
@@ -378,18 +378,30 @@ function completeFileArrayWithDatabaseInfo(&$filearray, $relativedir)
 				$rel_dir = preg_replace('/[\\/]$/', '', $rel_dir);
 				$rel_dir = preg_replace('/^[\\/]/', '', $rel_dir);
 
-				$ecmfile->filepath = $rel_dir;
-				$ecmfile->filename = $filename;
-				$ecmfile->label = md5_file(dol_osencode($filearray[$key]['fullname'])); // $destfile is a full path to file
-				$ecmfile->fullpath_orig = $filearray[$key]['fullname'];
-				$ecmfile->gen_or_uploaded = 'unknown';
-				$ecmfile->description = ''; // indexed content
-				$ecmfile->keywords = ''; // keyword content
-				$result = $ecmfile->create($user);
-				if ($result < 0) {
-					setEventMessages($ecmfile->error, $ecmfile->errors, 'warnings');
+				// verify if it's existing already
```

Comment:
When we are in this part, it means "We found a file called not indexed into database."
So we can add it. Why doing a fetch ? We should already no if it exists or not with the the if (!$found) line 363